addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -449,6 +449,12 @@ export const editorCommands = {
         if (closestEl && closestEl.getAttribute('contenteditable') === 'true') {
             editor._activateContenteditable();
         }
+        if (closestEl && closestEl.className.match(/_cta|btn|-[a-z]*/g)) {
+            closestEl.className = closestEl.className.replace(/_cta|btn|-[a-z]*|rounded|flat/g, '').trim();
+        }
```

Comment:
I think this is a nice try but the regexp is wrong as it will match things like `n-primary`. @qsm-odoo This is the thing that I mentionned the other day. I think I remember you had an opinion on this. How would you go about it ?

Basically the issue is that Firefox transform the `a` into a `span` when unlinking if it had any class, so we'd like to remove link related classes but keep the others as they might be used for styling that is unrelated to links.

Reply:
> I think this is a nice try but the regexp is wrong as it will match things like `n-primary`. @qsm-odoo This is the thing that I mentionned the other day. I think I remember you had an opinion on this. How would you go about it ?
> 
> Basically the issue is that Firefox transform the `a` into a `span` when unlinking if it had any class, so we'd like to remove link related classes but keep the others as they might be used for styling that is unrelated to links.

okay got it, updated the regex.

Reply:
@aath-odoo I think Quentin is right that the value of the behavior I asked for is not actually worth its added complexity. Do you still have the original version of the commit which completely removed the class ?

Reply:
@dmo-odoo okay, i've got it ,pushed changes.

Reply:
changes updated please check.

Reply:
changes updated please check.